src/test/java/io/cryostat/agent/FlightRecorderHelperTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.cryostat.agent.FlightRecorderHelper.TemplatedRecording;
+
+import jdk.jfr.Recording;
+import org.junit.jupiter.api.Test;
+
+public class FlightRecorderHelperTest {
+    @Test
```

Comment:
Could you make this a `@ParameterizedTest` and add cases for `default`, `Continuous`, `profile`, and `Profiling`? This way we can test that both of the standard OpenJDK built-in templates as well as our `ALL` "meta-template" all behave in the same way.